Cheza-Dzabala/way-farer
# Ch apply lfa feedback

#### What does this PR do?

Refactors code and enables users to book a specific number of seats. 
#### Description of Task to be completed?


1. Move all test models into a separate folder and must not be in the test files

2. Move all test files into a single test suite folder.

3. Import test files into an ` index.spec.js `

4. Update package.json to test the single `index.spec.js ` file.

5. Apply data validations to bookings, trips, and authentication. 

6. Fix response messages format.

7. Display to users number of seats available,

#### How should this be manually tested?

Clone the repo. 

Install node modules.

Create a trip.

Create a booking against newly created trip

Note that seat_availability on the trip is reduced. 

Using the admin account, cancel the booking. 

Note that seat availability is back up. 

#### Any background context you want to provide?

Users were unable to view seat availability

#### What are the relevant pivotal tracker stories?

###### #42167919179
#### Screenshots (if appropriate)

<img width="971" alt="Screenshot 2019-08-15 at 9 52 03 PM" src="https://user-images.githubusercontent.com/13886438/63122493-2f913580-bfa7-11e9-8ddb-c6eda7ee1bf4.png">
<img width="977" alt="Screenshot 2019-08-15 at 9 52 20 PM" src="https://user-images.githubusercontent.com/13886438/63122494-3029cc00-bfa7-11e9-83ff-b31bd61e368b.png">
<img width="810" alt="Screenshot 2019-08-15 at 9 53 25 PM" src="https://user-images.githubusercontent.com/13886438/63122495-3029cc00-bfa7-11e9-8382-c634b9a824f7.png">

#### Questions:
